### PR TITLE
[apache_tomcat]: Make parser configurable for catalina and localhost logs

### PR DESCRIPTION
**Pull request by ishleenk17**
- Enhancement

## Proposed commit message

- WHAT: The parsers have been made configurable for catalina and localhost logs
- WHY:  Apache tomcat can have logs through log4j which can be configured to support multiple date formats. Corresponding to that we need to provide support for multiple pattern for multiline parsers, hence providing this option to the customer to configure it as per need. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes https://github.com/elastic/integrations/issues/7204

## Screenshots

Screenshot 2023-10-26 at 4 50 45 PM

### CODE REVIEW

File: packages/apache_tomcat/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: 0.17.0
+  changes:
+    - description: Making parsers configurable for catalina and localhost logs
```
Comment (shmsr) [inline packages/apache_tomcat/changelog.yml:4]:
```suggestion
    - description: Making parsers configurable for catalina and localhost logs
```

File: packages/apache_tomcat/data_stream/catalina/manifest.yml [hunk 2]
```diff
@@ -45,6 +45,20 @@ streams:
         required: false
         show_user: false
         description: Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/fleet/current/elastic-agent-processor-configuration.html) for details.
+      - name: parsers
+        type: yaml
+        title: Parsers
+        description: |
+          This option expects a list of parsers that the payload has to go through. For more information see [Parsers](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-filestream.html#_parsers)
```
Comment (shmsr) [inline packages/apache_tomcat/data_stream/catalina/manifest.yml:52]:
```suggestion
          This option expects a list of parsers that the payload has to go through. For more information, see [Parsers](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-filestream.html#_parsers).
```

**Review by shmsr — CHANGES_REQUESTED** [review]

File: packages/apache_tomcat/data_stream/catalina/manifest.yml (hunk 2 above)
Comment (shmsr) [reply packages/apache_tomcat/data_stream/catalina/manifest.yml:52]: Similar changes in other data streams.

**Comment by shmsr** [comment]
PR title also needs update:

> [apache_tomcat]: Make parser configurable for catalina and localhost logs

**Comment by harnish-elastic** [comment]
@ishleenk17 If the user updates the multiline parser that we have supported, the logs are still dropped just because they might need to add their log date format in the [pipeline](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/data_stream/catalina/elasticsearch/ingest_pipeline/default.yml#L56) also. Can we capture these details in the README troubleshooting section for a better experience?

**Comment by ishleenk17** [comment]
> @ishleenk17 If the user updates the multiline parser that we have supported, the logs are still dropped just because they might need to add their log date format in the [pipeline](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/data_stream/catalina/elasticsearch/ingest_pipeline/default.yml#L29) also. Can we capture these details in the README troubleshooting section for a better experience?

Thats correct. The expectation is that we are leaving it upto the customer to have their own date formats through custom pipelines. Multi line parsers will help them to have those date formats being parsed for multiline logs.
But yes we can add this to the documentation.

**Review by harnish-elastic — APPROVED** [review]
LGTM! 🚀

**Review by shmsr — APPROVED** [review]

**Pull request merged**